### PR TITLE
filtered resources by tags

### PR DESCRIPTION
**Pull request by adityasimant**
## Fixes Issue
#18 

## Changes proposed

filtered resources by tags

### CODE REVIEW

**Comment by adityasimant** [comment]
``./awsctl get ec2 --tags=key,value1,value2`` this cmd will help you get ec2 instance using key values. You can have more than one values for a particular key.
``./awsctl get ec2 --tags=key`` If Values are not specified, it will show all values associated with that key and therefore will print those instances.

**Comment by surajnarwade** [comment]
@adityasimant can you update the commit title please?

File: cmd/ec2.go [hunk 1]
```diff
@@ -7,36 +7,66 @@ import (
...
 )
 
-// ec2Cmd represents the ec2 command
 var ec2Cmd = &cobra.Command{
 	Use:   "ec2",
 	Short: "Print ec2 related information",
 	Long: `For example,
 		awsctl get ec2`,
-	Run: getEC2command,
+	Run: func(cmd *cobra.Command, args []string) {
```
Comment (surajnarwade) [inline cmd/ec2.go:20]: keep as is and modify the `getEC2Command` function

File: pkg/getEc2Info.go [hunk 2]
```diff
@@ -9,50 +9,96 @@ import (
...
-	instanceName:="-"
+
+	instanceName := ""
+	var ec2Instance []EC2Instance
 	for _, r := range info.Reservations {
 		for _, i := range r.Instances {
-			for _,j:= range i.Tags{
-				if *j.Key == "Name"{
-					instanceName=aws.ToString(j.Value)
+			for _, j := range i.Tags {
```
Comment (surajnarwade) [inline pkg/getEc2Info.go:74]: Can we somehow break this into multiple functions? 
these nested for loops are not readable.

**Comment by riteshsonawane1372** [comment]
should we have a separate Client Function?

File: pkg/getEc2Info.go [hunk 3]
```diff
@@ -9,50 +9,96 @@ import (
...
+	InstanceType      string
+	InstancePrivateIP string
+}
 
-	InstanceName 		string
-	InstanceID	 		string
-	InstanceType 		string
-	InstancePrivateIP 	string
-
+func Ec2Client() (ec2.Client, context.Context) {
```
Comment (riteshsonawane1372) [inline pkg/getEc2Info.go:19]: We can have one Client function common to all

Comment (surajnarwade) [reply pkg/getEc2Info.go:19]: Common as in for particular group, for example EC2 will have instance,ami, EBS, etc

File: cmd/ec2.go [hunk 4]
```diff
@@ -7,36 +7,66 @@ import (
...
+func getEc2instanceTag(cmd *cobra.Command, args []string) {
+	var ec2Instance []pkg.EC2Instance
+	tags, _ := cmd.Flags().GetStringSlice("tags")
+
+	ec2Instance = pkg.GetEC2InstanceTag(&tags[0], tags[1:])
+
+	w := tabwriter.NewWriter(os.Stdout, 18, 5, 3, ' ', tabwriter.TabIndent)
+	defer w.Flush()
+
+	fmt.Fprintln(w, "TagValue", "\t", "INSTANCE_ID", "\t", "INSTANCE_TYPE", "\t", "PRIVATE IP")
```
Comment (surajnarwade) [inline cmd/ec2.go:41]: This should be TAG instead of TagValue and we are not printing tags on line 44

**Comment by surajnarwade** [comment]
@adityasimant you can see how it's done in kubectl: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

**Comment by riteshsonawane1372** [comment]
https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/label/label.go

**Pull request closed without merging**